### PR TITLE
Edit job backend

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
Implemented two APIs:
1. Update information about an existing job.
2. Delete an existing job.

### CODE REVIEW

**Comment by johan1505** [comment]
I will review this PR after you address the changes for the createJob one

File: src/server/src/routes/job.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            .end();
+    }
+});
+
+/******************************************************************************
+ *        DELETE Request example - Delete - "DELETE /api/job/delete/:id"
+ ******************************************************************************/
+
+router.delete('/delete/:id', async (req: jobRequest, res: Response) => {
+    const { id } = req.params;
```
Comment (johan1505) [inline src/server/src/routes/job.ts:182]: Add validation. If id is null then return an error response

File: src/server/src/routes/job.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    }
+});
+
+/******************************************************************************
+ *             POST Request example - Update - "POST /api/job/update"
+ ******************************************************************************/
+
+router.post('/update', async (req: jobRequest, res: Response) => {
+    const { job } = req.body;
+    const { targetYears,
```
Comment (johan1505) [inline src/server/src/routes/job.ts:120]: Add validation for the required fields that are deconstructed here. If either of them is missing, return an error response

File: src/server/src/routes/job.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        maxSalary,
+        departmentId } = job;
+    if (!job) {
+        return res.status(BAD_REQUEST).json({
+            error: errors.paramMissingError,
+        });
+    }
+    try {
+        // Check if required field is missing.
+        if (!targetYears || !hoursPerWeek || !description || !startDate || !type
```
Comment (johan1505) [inline src/server/src/routes/job.ts]: This if statement should be outside of the try block, right after the if (!job) statement.

**Comment by johan1505** [comment]
Left comments. Branch also have conflicts.

**Review by johan1505 — CHANGES_REQUESTED** [review]
Left comments. Branch also have conflicts.

File: src/server/src/routes/job.ts (hunk 1 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/job.ts:182]: Done.

File: src/server/src/routes/job.ts (hunk 3 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/job.ts]: Done.

File: src/server/src/routes/job.ts (hunk 2 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/job.ts:120]: Done.

**Review by johan1505 — APPROVED** [review]

**Pull request merged**